File: Iteration-2/src/Course.java

import java.util.ArrayList;

public class Course {
    private String courseID;
    private String courseName;
    private int credit;
    private ArrayList<Student> students;
    private String prerequisite;
    private int coursesemester;

    public Course(String courseID, String courseName, int credit, Lecturer courseLecturer,
                  ArrayList<Student> students, String prerequisite, int coursesemester) {
        this.courseID = courseID;
        this.courseName = courseName;
        this.credit = credit;
        this.students = new ArrayList<Student>();
        this.prerequisite = prerequisite;
        this.coursesemester = coursesemester;
    }

    public String getCourseID() {
        return courseID;
    }

    public void setCourseID(String courseID) {
        this.courseID = courseID;
    }

    public String getCourseName() {
        return courseName;
    }

    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }

    public int getCredit() {
        return credit;
    }

    public void setCredit(int credit) {
        this.credit = credit;
    }

    public ArrayList<Student> getStudents() {
        return students;
    }

    public void setStudents(Student student) {
        this.students.add(student);
    }

    public String getPrerequisite() {
        return prerequisite;
    }

    public int getCoursesemester() {
        return coursesemester;
    }
}

File: Iteration-3/Src/Course.py
class Course:
    def __init__(self, course_id, course_name, credit, lecturer, students, prerequisite, course_semester):
        self.course_id = course_id
        self.course_name = course_name
        self.credit = credit
        self.students = []
        self.prerequisite = prerequisite
        self.course_semester = course_semester

    def get_course_id(self):
        return self.course_id

    def set_course_id(self, course_id):
        self.course_id = course_id

    def get_course_name(self):
        return self.course_name

    def set_course_name(self, course_name):
        self.course_name = course_name

    def get_credit(self):
        return self.credit

    def set_credit(self, credit):
        self.credit = credit

    def get_students(self):
        return self.students

    def set_students(self, student):
        self.students.append(student)

    def get_prerequisite(self):
        return self.prerequisite

    def get_course_semester(self):
        return self.course_semester
